aptos-move/framework/aptos-framework/sources/permissioned_signer.move

```diff
@@ -0,0 +1,423 @@
+        assert!(
+            timestamp::now_seconds() < p.expiration_time,
+            error::permission_denied(E_PERMISSION_EXPIRED)
+        );
+        assert!(
+            exists<PermStorage>(p.permission_addr),
+            error::permission_denied(E_PERMISSION_REVOKED)
+        );
+        signer_from_permissioned_impl(p.master_addr, p.permission_addr)
+    }
+
+    public fun revoke_permission_handle(s: &signer, permission_addr: address) acquires GrantedPermissionHandles, PermStorage {
+        assert!(!is_permissioned_signer(s), error::permission_denied(ENOT_MASTER_SIGNER));
+        let master_addr = signer::address_of(s);
+        destroy_permission_address(permission_addr);
+        remove_permission_address(master_addr, permission_addr);
+    }
+
+    public fun revoke_all_handles(s: &signer) acquires GrantedPermissionHandles, PermStorage {
+        assert!(!is_permissioned_signer(s), error::permission_denied(ENOT_MASTER_SIGNER));
+        let master_addr = signer::address_of(s);
+        if(!exists<GrantedPermissionHandles>(master_addr)) {
+            return
+        };
+
+        let granted_permissions = borrow_global_mut<GrantedPermissionHandles>(master_addr);
+        let delete_list = vector::trim_reverse(&mut granted_permissions.active_handles, 0);
+        vector::destroy(delete_list, |address| {
+            destroy_permission_address(address);
+        })
+    }
+
+    public fun permission_address(p: &StorablePermissionedHandle): address {
+        p.permission_addr
+    }
+
+    public fun assert_master_signer(s: &signer) {
+        assert!(!is_permissioned_signer(s), error::permission_denied(ENOT_MASTER_SIGNER));
+    }
+
+    /// =====================================================================================================
+    /// Permission Management
+    ///
+
+    /// Authorizes `permissioned` with the given permission. This requires to have access to the `master`
+    /// signer.
+    public fun authorize<PermKey: copy + drop + store>(
+        master: &signer,
+        permissioned: &signer,
+        capacity: u256,
+        perm: PermKey
+    ) acquires PermStorage {
+        assert!(
+            is_permissioned_signer(permissioned) &&
+            !is_permissioned_signer(master) &&
+            signer::address_of(master) == signer::address_of(permissioned),
+            error::permission_denied(ECANNOT_AUTHORIZE)
+        );
+        let permission_signer = permission_signer(permissioned);
+        let permission_signer_addr = signer::address_of(&permission_signer);
+        let perms = &mut borrow_global_mut<PermStorage>(permission_signer_addr).perms;
+        let key = copyable_any::pack(perm);
+        if(simple_map::contains_key(perms, &key)) {
+            let entry = simple_map::borrow_mut(perms, &key);
+            entry.capacity = entry.capacity + capacity;
+        } else {
+            simple_map::add(perms, key, StoredPermission::V1 { capacity });
+        }
```

Comment:
I feel like we should have two different functions for setting capacity and incrementing amount. 

I would also make whole section of Permission Management to be public(friend) in this release, so that we can iterate on it later (and anyways we don't want third-party permissions for in v0)

Reply:
I we can only make the internal apis in the permissioned_signer.move to be public(package) as the scripts will need to invoke wrapper function to set up the permissions properly, and thus those functions will need to be public (e.g: FA permission setting function)

Reply:
I think we only need to keep one version that would increment for now, as this is the only one used. If there's further ask let's go implment that.